component/src/main/java/org/wso2/extension/siddhi/execution/geo/internal/impl/APIBasedGeoCoordinateResolver.java

```diff
@@ -0,0 +1,77 @@
+/*
+ *  Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.extension.siddhi.execution.geo.internal.impl;
+
+import org.apache.log4j.Logger;
+import org.wso2.extension.siddhi.execution.geo.api.GeoCoordinate;
+import org.wso2.extension.siddhi.execution.geo.api.GeoCoordinateResolver;
+import org.wso2.extension.siddhi.execution.geo.internal.utils.Utilities;
+import org.wso2.siddhi.core.exception.SiddhiAppRuntimeException;
+import org.wso2.siddhi.core.util.config.ConfigReader;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+
+/**
+ * The default implementation of the GeoCoordinateResolver interface. This implementation is based on "ipInfoDB" API.
+ */
+public class APIBasedGeoCoordinateResolver implements GeoCoordinateResolver {
+    private static final Logger LOGGER = Logger.getLogger(APIBasedGeoCoordinateResolver.class);
+    private String apikey;
+    private URL url;
+    private String locationDetails[];
+
+    @Override
+    public void init(ConfigReader configReader) {
+        apikey = configReader.readConfig("apiurl", "");
```

Comment:
Add validations

Reply:
"apiurl" can't validate here since without IP address, it's not a complete url. Later in the code I handled the MalformedURLException to check whether the URL is in a valid format.

Reply:
Null check has to be done since this a mandatory URL

Reply:
Fixed with commit : e4d056c2b32a587dad16015a2a866ed1ca529aa0